ow_regolith/src/RegolithSpawner.cpp

```diff
@@ -92,62 +68,55 @@ bool RegolithSpawner::initialize()
     ROS_ERROR("Regolith node requires the spawn_volume_threshold parameter.");
     return false;
   }
-  if (!m_node_handle->getParam("regolith_model_uri", m_model_uri)) {
+  string model_uri;
+  if (!m_node_handle->getParam("regolith_model_uri", model_uri)) {
     ROS_ERROR("Regolith node requires the regolith_model_uri parameter.");
     return false;
   }
-
-  // load SDF model
-  if (!getSdfFromUri(m_model_uri, m_model_sdf)) {
-    ROS_ERROR("Failed to load SDF for regolith model");
-    return false;
+  double spawn_spacing;
+  if (m_node_handle->getParam("spawn_spacing", spawn_spacing)) {
+    for (int i = 1;
+         i < static_cast<int>(floor(SCOOP_WIDTH / (2 * spawn_spacing)));
```

Comment:
It would be better imo to move this calculation and assigned to a (constant) variable right before the loop
```suggestion
         i < total_samples;
```

Reply:
Somehow missed this suggestion, but am now out of time for day, so I'll get on it tomorrow.

Reply:
Done.